src/core_plugins/kibana/server/routes/api/suggestions/register_value_suggestions.js

```diff
@@ -4,33 +4,40 @@ export function registerValueSuggestions(server) {
   server.route({
     path: '/api/kibana/suggestions/values/{index}',
     method: ['POST'],
-    handler: function (req, reply) {
+    handler: async function (req, reply) {
       const { index } = req.params;
       const { field, query } = req.payload;
-
       const { callWithRequest } = server.plugins.elasticsearch.getCluster('data');
-      const include = query ? `.*${query}.*` : undefined;
-      const body = getBody({
-        field,
-        include,
-        shard_size: 10,
-        size: 10
-      });
-
-      return callWithRequest(req, 'search', { index, body })
-      .then((res) => {
-        const suggestions = res.aggregations.suggestions.buckets.map(bucket => bucket.key);
-        return reply(suggestions);
-      })
-      .catch(error => reply(handleESError(error)));
+      const body = getBody({ field, query });
+      try {
+        const response = await callWithRequest(req, 'search', { index, body });
+        const suggestions = response.aggregations.suggestions.buckets.map(bucket => bucket.key);
+        reply(suggestions);
+      } catch (error) {
+        reply(handleESError(error));
+      }
     }
   });
 }
 
-function getBody(terms) {
+function getBody({ field, query }) {
   return {
+    size: 0,
+    timeout: '1s',
+    terminate_after: 100000,
     aggs: {
-      suggestions: { terms }
+      suggestions: {
+        terms: {
+          field,
+          include: `${getEscapedQuery(query)}.*`,
+          execution_hint: 'map'
```

Comment:
also since we do not care about the accuracy of the counts, we could set the `shard_size`to 10 explicitly in order to reduce the amount of information that needs to be transmitted from shards to the coordinating node